Zend/zend_attributes.c

```diff
@@ -193,6 +194,29 @@ ZEND_METHOD(Deprecated, __construct)
 	}
 }
 
+ZEND_METHOD(NoDiscard, __construct)
+{
+	zend_string *message = NULL;
+	zval value;
+
+	ZEND_PARSE_PARAMETERS_START(0, 1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_STR_OR_NULL(message)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (message) {
+		ZVAL_STR(&value, message);
+	} else {
+		ZVAL_NULL(&value);
+	}
+	zend_update_property_ex(zend_ce_nodiscard, Z_OBJ_P(ZEND_THIS), ZSTR_KNOWN(ZEND_STR_MESSAGE), &value);
+
+	/* The assignment might fail due to 'readonly'. */
```

Comment:
I find code like this noisy, this just checks the contract of RETURN_THROWS. I would prefer to just keep the comment and remove the check

Reply:
I'm not sure I follow. Do you mean “remove the check” (thus keeping the `RETURN_THROWS()`)? This would be incorrect if the assignment doesn't fail.

Or do you mean “remove the entire block, including the `RETURN_THROWS()`”? In that case the comment no longer makes sense.

In any case, this mirrors the constructor of `Deprecated::__construct()`, which has 2 readonly properties and where the `RETURN_THROWS()` is actually doing something, so I find it helpful for clarity. In production builds this should hopefully not result in any emitted assembly either.

see also https://github.com/php/php-src/pull/11293#discussion_r1642951378 and https://github.com/php/php-src/pull/11293#discussion_r1661525090.